src/data/list/perm.lean

```diff
@@ -1052,6 +1061,53 @@ end
 @[simp] theorem mem_permutations (s t : list α) : s ∈ permutations t ↔ s ~ t :=
 ⟨perm_of_mem_permutations, mem_permutations_of_perm_lemma mem_permutations_aux_of_perm⟩
 
+theorem perm_permutations'_aux_comm (a b : α) (l : list α) :
+  (permutations'_aux a l).bind (permutations'_aux b) ~
+  (permutations'_aux b l).bind (permutations'_aux a) :=
+begin
+  induction l with c l ih, {simp [swap]},
+  simp [permutations'_aux], apply perm.swap',
+  have : ∀ a b,
+    (map (cons c) (permutations'_aux a l)).bind (permutations'_aux b) ~
+    map (cons b ∘ cons c) (permutations'_aux a l) ++
+    map (cons c) ((permutations'_aux a l).bind (permutations'_aux b)),
+  { intros,
+    simp [map_bind, permutations'_aux],
+    refine (bind_append_perm _ (λ x, [_]) _).symm.trans _,
+    rw [← map_eq_bind, ← bind_map] },
+  refine (((this _ _).append_left _).trans _).trans ((this _ _).append_left _).symm,
+  rw [← append_assoc, ← append_assoc],
+  exact perm_append_comm.append (ih.map _),
+end
+
+theorem perm.permutations' {s t : list α} (p : s ~ t) :
+  permutations' s ~ permutations' t :=
```

Comment:
How hard is it to prove the other direction too? There's absolutely no need to do so in this PR, but it would be a nice result to have.

Reply:
It's easy. `permutations'` is a perm of `permutations` so we know that `mem_permutations` also holds for `permutations'`. So `s \in permutations s ~ permutations t` implies `s ~ t`.

Reply:
Can you add the proof so that `permutations' s ~ permutations' t ↔ s ~ t` can be a `simp` lemma?